src/components/editor/index.js

```diff
@@ -0,0 +1,261 @@
+import { PropTypes } from 'react';
+import reactStamp from 'react-stamp';
+import { fromJS, List, Repeat } from 'immutable';
+import Editor from 'draft-js-plugins-editor';
+import {
+  EditorState,
+  ContentState,
+  ContentBlock,
+  convertFromHTML,
+  convertToRaw,
+  convertFromRaw,
+  RichUtils,
+  Modifier,
+  BlockMapBuilder,
+  genKey,
+  CharacterMetadata,
+} from 'draft-js';
+
+function insertFragment(editorState, fragment, entityMap) {
+  var newContent = Modifier.replaceWithFragment(
+    editorState.getCurrentContent(),
+    editorState.getSelection(),
+    fragment
+  );
+
+  return EditorState.push(
+    editorState,
+    newContent,
+    'insert-fragment'
+  );
+}
+
+const stripPastePlugin = {
+  handlePastedText(text, html, { getEditorState, setEditorState }) {
```

Comment:
This plugin should have its own unit tests. This is the reason it needed to be a plugin. For in that world, we can have a "strip paste plugin factory". A unit test tests a "unit". In this case, that our stripping is correctly working. We assume all other components are working per spec - including `convertFromHTML`.

Ergo, if we pass the conversion function into the plugin creation factory, we can just pass in a mock in our test - because we're not trying to test Draft's code. And therefore, the original "pasted" HTML doesn't matter either. We just pass a mock `convertFromHTML` function that returns a pre-defined content block that we ensure gets appropriately stripped before passing to `setEditorState`.

wdyt?